test/vinyl-luatest/gh_10233_index_build_vs_snapshot_test.lua

```diff
@@ -0,0 +1,41 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new()
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.after_each(function(cg)
+    cg.server:exec(function()
+        if box.space.test ~= nil then
+            box.space.test:drop()
+        end
+    end)
+end)
+
+g.test_index_build_vs_snapshot = function(cg)
+    cg.server:exec(function()
+        local fiber = require('fiber')
+        local s = box.schema.space.create('test', {engine = 'vinyl'})
+        s:create_index('pk')
+        s:insert({1, 1})
+        local f1 = fiber.new(s.create_index, s, 'sk', {parts = {2, 'unsigned'}})
+        f1:set_joinable(true)
+        local f2 = fiber.new(box.snapshot)
+        f2:set_joinable(true)
+        fiber.sleep(0.1)
+        local ok, err
+        local timeout = 5
+        ok, err = f1:join(timeout)
+        t.assert_equals(ok or err, true)
```

Comment:
Nit. Rather cryptic. May be just `t.assert(ok, err)`.

Reply:
AFAIU the message shouldn't depend on the test case result so I wouldn't use it. To make the code more straightforward changed to this:

```diff
diff --git a/test/vinyl-luatest/gh_10233_index_build_vs_snapshot_test.lua b/test/vinyl-luatest/gh_10233_index_build_vs_snapshot_test.lua
index ac0691b26a75..3a9ddec74345 100644
--- a/test/vinyl-luatest/gh_10233_index_build_vs_snapshot_test.lua
+++ b/test/vinyl-luatest/gh_10233_index_build_vs_snapshot_test.lua
@@ -26,16 +26,17 @@ g.test_index_build_vs_snapshot = function(cg)
         local s = box.schema.space.create('test', {engine = 'vinyl'})
         s:create_index('pk')
         s:insert({1, 1})
-        local f1 = fiber.new(s.create_index, s, 'sk', {parts = {2, 'unsigned'}})
+        local f1 = fiber.new(function()
+            s:create_index('sk', {parts = {2, 'unsigned'}})
+        end)
         f1:set_joinable(true)
-        local f2 = fiber.new(box.snapshot)
+        local f2 = fiber.new(function()
+            box.snapshot()
+        end)
         f2:set_joinable(true)
         fiber.sleep(0.1)
-        local ok, err
         local timeout = 5
-        ok, err = f1:join(timeout)
-        t.assert_equals(ok or err, true)
-        ok, err = f2:join(timeout)
-        t.assert_equals(ok or err, true)
+        t.assert_equals({f1:join(timeout)}, {true})
+        t.assert_equals({f2:join(timeout)}, {true})
     end)
 end
```